tutorials/epochs/baseline_regression.py

```diff
@@ -0,0 +1,185 @@
+# -*- coding: utf-8 -*-
+"""
+.. _ex-baseline_regression:
+
+====================================================================
+Baseline correction using generalized linear mixed models
+====================================================================
+
+By including the baseline interval as a regressor in a generalized
+linear model, we allow the data to determine how much baseline
+correction is necessary for the specific epoch. This approach can
+increase statistical power (Alday et al., 2019).
+Alday et al. provided a script at osf: https://osf.io/85bjq
+"""
+# Authors: Carina Forster
+#
+# License: BSD-3-Clause
+
+import numpy as np
+import matplotlib.pyplot as plt
+
+import mne
+from mne.datasets import sample
+
+# %%
+
+#we load the example dataset for one subject
+
+data_path = sample.data_path()
+
+subjects_dir = data_path / 'subjects'
+meg_path = data_path / 'MEG' / 'sample'
+raw_fname = meg_path / 'sample_audvis_filt-0-40_raw.fif'
+
+#define epoch length
+
+tmin, tmax = None, 0.500
+
+# %% load the raw data for one subject
+
+raw = mne.io.read_raw_fif(raw_fname, preload=True)
+
+# we are only interested in EEG data
+
+raw.pick_types(stim=True, eog=True, exclude=(), eeg=True)
+
+# we filter the data (similar to Alday et al.)
+
+lowpass = 40
+highpass = 0.5 # Alday et al used 0.1
```

Comment:
The choice of high-pass filter will influence how much baseline is actually used, so I would motivate the choice of filter here.